app/views/2017/shared/_header.html.erb

```diff
@@ -16,9 +16,9 @@
       <li class="nav-link"><%= link_to t("header.about"),   "/about" %></li>
     </ul>
 
-    <% unless Locale.english? %>
+    <% unless Current.locale.english? %>
       <ul>
-        <li class="nav-link"><%= link_to t('header.locale_articles'), language_path(@site_locale.canonical) %></li>
+        <li class="nav-link"><%= link_to t('header.locale_articles'), language_path(Current.locale.name) %></li>
```

Comment:
I don't know if this is the source of the test failures, but `canonical` == `slug`

```suggestion
        <li class="nav-link"><%= link_to t('header.locale_articles'), language_path(Current.locale.slug) %></li>
```